impress-org/givewp
# Feature: add alternating banners by pricing tier

Blocked: #7408

Resolves: [GIVE-872] & [GIVE-873]

## Description
This PR addresses the implementation of two sets of banners: one for our core plugin and another for P2P. Each set contains three variations based on the user's pricing tier/plan. The primary goal is to:

- Provide additional banners: Include the methods and data necessary to display banners based on users pricing tier/plan.
- Alternate Banners: Introduce a mechanism to alternate between the core plugin banner and the P2P banner. 

Only one banner will be displayed at a time on the Donor, Donation and Reports page. Each banner ID can be closed separately and should remain closed. However the main change is that if both the Give & P2P banners are available, anytime a user visits one of the specified pages they will see the a different banner from the previous one. These alterations are being tracked in a session.

### Banner information: 
[Figma](https://www.figma.com/design/nIQJs7W7vNJ6WZxnX2crCj/Stellar-Sale-2024-banners?node-id=0-1&t=W99obUh8ls6WNw7i-0)

Get 40% off Plus Plan: https://go.givewp.com/plusplan
Get 40% off StellarWP: https://go.givewp.com/stellarsale
Get 40% off P2P (Single): https://go.givewp.com/p2p
Get 40% of P2P (Plan): https://go.givewp.com/p2ppro
Get 40% off Pro: https://go.givewp.com/pro

## Affects
Donations, Donors, Report pages

## Visuals


## Testing Instructions
- adjust the date-range in the StellarSaleBanner class inside the `getBanners` method, this should enable the banner.
- visit the Donation, Donor & Report pages to verify that a banner is displayed. Take note of which banner it is.
- visit the other pages, each visit should alternate which banner you see on display.
- verify that the links of each banner work
- close a banner by clicking the icon in the top right and verify it remains closed on all 3 pages.
- after closing a single banner it should no longer be alternating.
- close both banners.

## Pre-review Checklist



[GIVE-872]: https://stellarwp.atlassian.net/browse/GIVE-872?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[GIVE-873]: https://stellarwp.atlassian.net/browse/GIVE-873?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
